### ISSUE TITLE
Database update issues and we need to upgrade to MW 1.25+ 

### ISSUE THREAD

**Issue by Arcane21** [Orain/ansible-playbook#690]
On July 27, 2015 11.00PM I attempted to change the ext branch for some of our localisation exts to the master branch since the 1.24 ones have serious regressions that have been causing recurring issues. This appeared successful, but I had to wait and see if we'd have another LC cache degradation events.

On July 28, 2015 6:30AM we did, since I wanted to make sure I hadn't missed anything, I changed the rest of the MLEB bundle exts and a few other that had recurring magic work/system message breakdown issues to master to see if that would fix our issues.

They turned out to require MediaWiki 1.25+, so I reverted all the changes and immediately ran an ansible update to make sure this didn't crap out the entire farm.

Unfortunately, the databases now refuse to update via shell, doing an ansible run nor running the standard database update is working, and since ansible is working fine, our database servers appear to be down for reasons I cannot determine, as it appears ansible cannot even access them at all.

Once we get them back up, it's becoming obvious we need to upgrade to MediaWiki 1.25 or the trunk branch, since many extensions have regressions only these branches fix that we need to prevent issues. I know MW 1.25+ uses a new extension load schema that gave us hassles last time, but we need to adapt it PDQ, because our current setup is not working very well while we remain on 1.24.

**Comment by addshore** [comment]
So a quick note I made this change https://github.com/Orain/ansible-playbook/commit/7701cd6509514d932266ed15c647adff04420a33 meaning that in future when the site crashes the db lists should not break in the same way it did here

**Comment by addshore** [comment]
And a further not on the issue as a whole.
Yes, if you try to update extensions that need mw 1.25+ when the site is only running 1.26 everything is going to break ;)
MW must be updated first!

This is also why we should avoid pushing potentially breaking things straight to master or merging our own pull requests in such cases...

**Comment by Arcane21** [comment]
SPF explained the db.list breakage to me, that was somethign I had no clue about until he did so.

As for the extensions, natch, but for some we've had to set them to master because the branch versions just don't work as well, and in this case, I was trying to overcome currently known LC cache issues with the branch versions we have based on information gathered from MW.org, Phabricator, and the WMF repo concerning the issues.

We still need to upgrade to 1.25+ ASAP in any event, and I agree entirely it needs to be done at once, not piecemeal.

**Comment by Arcane21** [comment]
I going to reiterate, with a little more force this time, we NEED to upgrade to MW 1.25, even to the development branch of WMF again.

I checked, a lot of currently busted extension have been fixed in 1.25+, including a lot of LC issues that have been recurring.

Also, see this:

http://drunkardswalkforums.yuku.com/reply/131449/All-The-Tropes-Wiki-Project-Part-V#reply-131449

Vector is a POS for many users on ATT, with many features, gadgets, and interface features like CollapsibleVector being a crap shoot for many users, myself included. It apparently only affects large wikis, smaller ones seem to be fine (examination of Pabricator/WMF repo commit logs appear to bear this out), but since ATT is still our main draw, on behalf of our userbase, we want this fixed PDQ.

I know MW 1.25+ has a new extension load scheme that is annoying, but the longer we stay with 1.24, the longer we have all these issues, and frankly, I'm getting sick of it like a lot of others are.

**Comment by addshore** [comment]
Arcane, if you let me know which extensions and exact fixes we need I would probably get them backported to the release we are currently using (as a shortterm solution)

PDQ?

But yes we do also need to upgrade MW.
In the case of upgrading mediawiki...
 - Mediawiki needs to be updated first
  - We need to make sure that patches such as https://github.com/Orain/mediawiki-core/commit/f1e5359d06f8186d05902e0ea6aad71d4a1f9558 are copied over to our new version of core.
  - We need to READ THE CHAGELOG and make sure nothing is going to break
- Then update chunks of extensions testing as we go..

**Comment by Arcane21** [comment]
Here is a short list of extensions currently affected:

LiquidThreads, BetaFeatures, Echo, OnlineStatus, NoTitle, Translate (and related MLEB ext), RecentChangesLogFilter

Of that list, the first three ext and the MLEB ones are the most crippled.

Also, good rebuke on my oversights, I stand corrected.

**Comment by Arcane21** [comment]
At the risk of sounding downright bitchy, I need that autorun LC refresh for all MW servers set up to run without human need to do it manually YESTERDAY.

See here for details:

https://allthetropes.orain.org/wiki/Forum:Tech_Wishlist_And_Bug_Reports#Templates_3501

I cannot be available at all hours of the day to fix this manually until we finally get an upgrade done, so could someone PLEASE get this done?

I've been asking for over a month now.

**Comment by addshore** [comment]
Running LC cache every hour is not a solution really >.>
Again, we need to find the problem.. It makes no sense that the LC cache keeps falling apart.. we need to find the cause.

As for the extensions you said there are fixes on master for things you need.
If you link me to these fix commits I can more than likely put them on the branch for the version of MW we are still using and thus fix those extension problems for you! But you need to give me links!

Also Assigning random people to this issue will probably not get it solved any quicker.

And also this is not just affecting ATT, but all wikis on the farm, I would love to spend an hour or so looking into why exactly this is happening but I honestly do not have any free time currently....
If you need any help looking into the root cause of the issue itself feel free to ask howto go about it / email me etc...

Also this issue is not to do with LC cache but to do with upgrading MW, so it should probably be discussed in another ticket to keep all relevant information together.....

Again, sorry this is not fixed...

**Comment by Arcane21** [comment]
Backporting the fixes is not as easy as just editing a few PHP files. MW 1.25+ uses a completely new Extension loader schema, so we need to do a wholesale update.

Also, the auto LC cache refresh is not meant to be a permanent fix, more a stopgap until we can hunt down the true cause, but it would take a lot of hassle off us while we hunt it down.

As for the multiple assignments, I have no idea HOW to set up an auto running job on all MW servers every hour, hence my hoping someone more skilled could set that up to excise some of the hassle until we implement a more permanent solution.

**Comment by addshore** [comment]
Yes but the fixes to bugs should basically have nothing to do with the loading of the extension, and again even if it does the fix can be backported to the branch for the older version.

If you have the time you should be able to add a PR to add a cronjob to the MW servers for referesh the LC cache!

To see all other crons for the mediawiki servers take a look at https://github.com/Orain/ansible-playbook/blob/master/roles/mediawiki/tasks/cron.yml
Feel free to ask me anything!

**Comment by Arcane21** [comment]
Well, backporting is beyond my expertise.

As for making that cron job job, thanks, I'm going to go over some notes I made on the degeneration of the LC cache to figure out how often the cron job needs to run, then I'll submit a pull request.

**Comment by addshore** [comment]
you dont need to know how to backport! just link me to the fixes you need!!!!

**Comment by Arcane21** [comment]
Alright, will take awhile to dig them up, there are a lot.
